crates/sncast/src/starknet_commands/script.rs

```diff
@@ -452,17 +337,53 @@ pub fn run(
     // import from cairo-lang-runner
     let (hints_dict, string_to_hint) = build_hints_dict(instructions.clone());
 
-    let mut cairo_hint_processor = CairoHintProcessor {
+    // hint processor
+    let block_context = build_block_context(CheatnetBlockInfo::default());
+    let account_context = build_transaction_context();
+    let mut context = EntryPointExecutionContext::new(
+        &block_context.clone(),
+        &account_context,
+        ExecutionMode::Execute,
+        false,
+    )
+    .unwrap();
+
+    let mut blockifier_state = CachedState::from(DictStateReader::default());
+    let mut execution_resources = ExecutionResources::default();
+
+    let syscall_handler = SyscallHintProcessor::new(
+        &mut blockifier_state,
+        &mut execution_resources,
+        &mut context,
+        // This segment is created by SierraCasmRunner
+        Relocatable {
+            segment_index: 10,
+            offset: 0,
+        },
+        CallEntryPoint::default(),
+        &string_to_hint,
+        ReadOnlySegments::default(),
+    );
+
+    let cast_extension = CastExtension {
         hints: &string_to_hint,
         provider,
-        runtime,
-        run_resources: RunResources::default(),
+        tokio_runtime,
         config,
     };
 
-    match runner.run_function(
+    let mut cast_runtime = ExtendedRuntime {
+        extension: cast_extension,
+        extended_runtime: StarknetRuntime {
+            hint_handler: syscall_handler,
+        },
+    };
+
+    let mut vm = VirtualMachine::new(true);
+    match runner.run_function_with_vm(
```

Comment:
I believe in the future we may want to have the `execute` function that is generalised too (traits etc.) to make sure we don't do the same work twice. For example now we change this place to use casm runner in #1206
cc @cptartur @MaksymilianDemitraszek

Reply:
yes sir